### PR TITLE
Download header files from the gallery

### PR DESCRIPTION
**Pull request by ahms5**
### Changes proposed in this pull request:

~~- requires https://github.com/pyfar/gallery/pull/50~~
- ~~download the navbar, stylefiles and logos which where copied from gallery.~~
- ~~remove redundancies~~

~~after https://github.com/pyfar/gallery/pull/50 is merged we need to change the branch to main~~

latest changes: (2024-03-24)
- intoduce pyfar as subpage to create left sidebar and dymanic header
- download all the static header stlyes logos, stylefiles from gallery automatically
    - '_static/css/custom.css',
    - '_static/favicon.ico',
    - '_static/header.rst',
    - 'resources/logos/pyfar_logos_fixed_size_pyfar.png'
- requires https://github.com/pyfar/gallery/pull/52

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -125,6 +127,29 @@
    "default_mode": "light"
 }
 
+# -- download navbar and style files from gallery -----------------------------
+branch = 'static-nav-bar'
```
Comment (f-brinkmann) [inline docs/conf.py:131]: This will be changed, once the pull in the gallery is merged, right?

File: docs/conf.py [hunk 2]
```diff
@@ -125,6 +127,29 @@
...
+    '_static/css/custom.css',
+    '_static/favicon.ico',
+    '_templates/navbar-nav-static.html',
+    'resources/logos/pyfar_logos_fixed_size_pyfar.png'
+    ]
+c = urllib3.PoolManager()
+for file in folders_in:
+    url = link + file
+    filename = file
+    print(url)
```
Comment (f-brinkmann) [inline docs/conf.py:143]: Do we need this or was it only for debugging?

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Generally approved, but to small comments

File: docs/conf.py (hunk 1 above)
Comment (ahms5) [reply docs/conf.py:131]: yes.
https://github.com/pyfar/gallery/pull/50

File: docs/pyfar.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+   history
+
+..
+   (removed because it is contained in the new spinx style)
+   Indices and tables
+   ==================
+   * :ref:`genindex`
+   * :ref:`modindex`
+   * :ref:`search`
```
Comment (sikersten) [inline docs/pyfar.rst:43]:
```suggestion
```

**Comment by sikersten** [comment]
Looks good to me, thanks for taking care! What I do not understand: why was the former content of index.rst moved to pyfar.rst, and then index.html is replaced with pyfar.html? Why can't header.rst just be included at the top in index.rst?

File: .gitignore [hunk 4]
```diff
@@ -124,3 +124,10 @@ pyfar/signals/files/*
 # workaround for failing test discovery in vscode
 tests/*/__init__.py
 tests/private/
+
+# ingore doc stuff donloaded from pyfar gallery
```
Comment (sikersten) [inline .gitignore:128]:
```suggestion
# ignore downloads from pyfar gallery
```

**Review by sikersten — APPROVED** [review]
Thanks, I get it now! Approved!

**Comment by ahms5** [comment]
if we directly merge to main and rebuild the doc on readthedocs, we dont need a wait until the release

**Review by sikersten — APPROVED** [review]

File: docs/conf.py [hunk 5]
```diff
@@ -107,6 +110,7 @@
     "navbar_start": ["navbar-logo"],
     "navbar_end": ["navbar-icon-links", "theme-switcher"],
     "navbar_align": "content",
+    "header_links_before_dropdown": 8,
```
Comment (pingelit) [inline docs/conf.py:113]: This was 10 in the gallery. Is this by design?

**Review by pingelit — APPROVED** [review]
Thanks, great change. That makes the pyfar page a lot more usable in my opinion with the side bar.
One minor comment, but I might not be getting the complete picture.

File: docs/conf.py (hunk 5 above)
Comment (ahms5) [reply docs/conf.py:113]: I think 8 is by definition no ``More`` Dropdown

**Comment by ahms5** [comment]
replaced by #569

**Pull request closed without merging**